Data/Aeson/Types/Instances.hs

```diff
@@ -182,6 +182,21 @@ instance FromJSON Bool where
     parseJSON = withBool "Bool" pure
     {-# INLINE parseJSON #-}
 
+instance ToJSON Ordering where
+  toJSON o = String $ case o of
+    LT -> "LT"
+    EQ -> "EQ"
+    GT -> "GT"
+
+instance FromJSON Ordering where
+  parseJSON v = case v of
+    String s -> case s of
```

Comment:
Could you replace the explicit case analysis on `v`with [withText](http://hackage.haskell.org/package/aeson-0.10.0.0/docs/Data-Aeson-Types.html#v:withText). That's a bit shorter and consistent with other text-based encodings.